src/onegov/search/mixins.py

```diff
@@ -134,6 +136,139 @@ def es_tags(self):
         """ Returns a list of tags associated with this content. """
         return None
 
+    @property
+    def search_score(self):
+        """
+        the lower the score they higher the class type will be shown in search
+        results. Default is 10 (lowest)
+        """
+        return 10
+
+    def __lt__(self, other):
+        return self.search_score < other.search_score
+
+    @staticmethod
+    def psql_tsvector_string(model):
+        """
+        Provides the tsvector string for postgres defining which columns and
+        json fields to be used for full text search index.
+
+        Example:
+        s = create_tsvector_string('title', 'location')
+        s += " || ' ' || coalesce(((content ->> 'description')), '')"
+        s += " || ' ' || coalesce(((content ->> 'organizer')), '')"
+
+        :return: tsvector string
+        """
+
+        join = " || ' ' || "
+
+        # identify search columns
+        columns = [p for p in model.es_properties if p in model.__dict__
+                   and isinstance(model.__dict__[p], InstrumentedAttribute)]
+        s = Searchable.create_tsvector_string(*columns) if columns else ''
+
+        # identify content and meta properties
+        for p in model.es_properties:
+            if p in model.__dict__ and isinstance(model.__dict__[p],
+                                                  content.dict_property):
+                s += join if s else ''
+                s += f"coalesce((({model.__dict__[p].attribute} " \
+                     f"->> '{p}')), '')"
+
+        return s
+
+    @staticmethod
+    def reindex(session, schema, model):
+        """
+        Re-indexes the table by dropping and adding the full text search
+        column.
+        """
+        Searchable.drop_fts_column(session, schema, model)
+        Searchable.add_fts_column(session, schema, model)
```

Comment:
Since `add_fts_column` needs to be more sophisticated to be correct it would probably be a good idea to assert that the model that was passed in is either not polymorphic, or a polymorphic base class.

You could also just do the same assertion in `drop_fts_column` and `add_fts_column`. Although `drop_fts_column` can afford to be a little bit more forgiving.